src/legacy/core_plugins/telemetry/common/get_xpack_config_with_deprecated.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { KibanaConfig } from 'src/legacy/server/kbn_server';
+
+export function getXpackConfigWithDeprecated(config: KibanaConfig, configPath: string) {
```

Comment:
have we considered using the deprecations property, [example](https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/security/index.js#L60), there's a rename function passed in

Reply:
I am using the deprecations property in the `xpack_main` plugin to tell users that `xpack.xpack_main.telemetry.*` configs are deprecated. This function only attempts to grab the config from the deprecated flags. If I use the rename function in `deprecations` does it automatically solve this?

Reply:
yep it should

Reply:
So i've checked on this one; the rename function only works for same plugin renames. So I cannot change `xpack.xpack_main.telemetry.*` to `telemetry.*`. I can only change something like `xpack.xpack_main.telemetry.x` to `xpack.xpack_main.telemetry.y` assuming i am in the `xpack_main` plugin.

Reply:
👍